content/docs/rfcs/12/README.md

```diff
@@ -61,77 +63,150 @@ The following are not considered as part of the adversarial model:
+
+Since a filter service node is consuming resources to provide this service,
+it MAY account for usage and adapt its service provision to certain clients.
+An incentive mechanism is currently planned but underspecified.
+
+### Filter Subscribe Request
+
+A client node MUST send all filter requests in a `FilterSubscribeRequest` message.
+This request MUST contain a `request_id`.
+The `request_id` MUST be a uniquely generated string.
+Each request MUST include a `filter_subscribe_type`, indicating the type of request.
+
+### Filter Subscribe Response
+
+In return to any `FilterSubscribeRequest`,
+a filter service node SHOULD respond with a `FilterSubscribeResponse` with a `requestId` matching that of the request.
+This response MUST contain a `status_code` indicating if the request was successful or not.
+Successful status codes are in the `2xx` range.
+Client nodes SHOULD consider all other status codes as error codes and assume that the requested operation had failed.
+In addition, the filter service node MAY choose to provide a more detailed status description in the `status_desc` field.
+
+### Filter matching
+
+In the description of each request type below,
+the term "filter criteria" refers to the combination of `pubsub_topic` and a set of `content_topics`.
+The request MAY include filter criteria, conditional to the selected `filter_subscribe_type`.
+The `pubsub_topic` is always optional in filter criteria.
+A `WakuMessage` matches filter criteria when its `content_topic` is in the `content_topics` set
+and, if included in the criteria, it was published on a matching `pubsub_topic`.
+If the `pubsub_topic` is set and not empty,
+a `WakuMessage` will only match the filter criteria if it has a matching `content_topic` and was published on _the same_ `pubsub_topic`.
+If the `pubsub_topic` is either empty or unset,
+a `WakuMessage` will match the filter criteria if it has a matching `content_topic` and was published on _any_ `pubsub_topic`.
+
+### Filter Subscribe Types
+
+The following filter subscribe types are defined:
+
+#### SUBSCRIBER_PING
+
+A filter client that sends a `FilterSubscribeRequest` with `filter_subscribe_type` set to `SUBSCRIBER_PING`
+requests that the service node SHOULD indicate if it has any active subscriptions for this client.
+The filter client SHOULD exclude any filter criteria from the request.
+The filter service node SHOULD respond with a success code if it has any active subscriptions for this client
+or an error code if not.
+
+#### SUBSCRIBE
+
+A filter client that sends a `FilterSubscribeRequest` with `filter_subscribe_type` set to `SUBSCRIBE`
+requests that the service node SHOULD push messages matching this filter to the client.
+The filter client MUST include the desired filter criteria in the request.
+A client MAY use this request type to _modify_ an existing subscription
+by providing _additional_ filter criteria in a new request.
```

Comment:
Yes! This is the plan in terms of how it will be implemented.

After trying to settle on terminology I came up with the earlier stated:

> The combination of different filter criteria for a specific filter client node is termed a "subscription".

Each filter criterion/transaction will indeed be modelled as a "pubsub_topic+content_topic" which can only be added or removed. I found that people most often spoke about that action as "updating my subscription", but am not set on the terminology.